### ISSUE TITLE
PEAR/FunctionCallSignature: new bug introduced via #2506

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#2647]
The fix for issue #2506 has introduced a new bug where code which is formatted perfectly fine, now triggers errors.

Simplest code pattern I could find to reproduce the issue:
```php
$notices = array(
	'index' => sprintf(
		translation_function('a text string with %s placeholder'),
		'replacement'
	),
);
```

I think it may have something to do with function calls within array declarations as all the instances I've seen so far have involved those.

Command to reproduce:
```bash
phpcs -ps ./file.php --standard=PEAR --sniffs=PEAR.Functions.FunctionCallSignature --tab-width=4
```

Unexpected errors - no errors should be thrown, this code is clean:
```
FILE: I:\000_GitHub\PHPCS\Tests\phpcs-xxx.php
------------------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
------------------------------------------------------------------------------------------
 5 | ERROR | [x] Multi-line function call not indented correctly; expected 4 spaces but
   |       |     found 8 (PEAR.Functions.FunctionCallSignature.Indent)
 6 | ERROR | [x] Multi-line function call not indented correctly; expected 4 spaces but
   |       |     found 8 (PEAR.Functions.FunctionCallSignature.Indent)
 7 | ERROR | [x] Multi-line function call not indented correctly; expected 0 spaces but
   |       |     found 4 (PEAR.Functions.FunctionCallSignature.Indent)
------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------
```

**Comment by gsherwood** [comment]
Thanks a lot for finding and reporting this. Was an issue when the array index was a string - the sniff was trying to account for multi-line strings but I forgot about that case.

Any chance you can give the sniff another test?

**Comment by jrfnl** [comment]
Tested and on the project where I found the issue all looks fine.

If I find additional issues at a later point, I will open a new issue.

**Comment by gsherwood** [comment]
Thanks a lot for retesting this.
